### PR TITLE
Set resource requests/limits for TargetAllocator

### LINKED ISSUE #1103: set resource requests/limits for target-allocator
Our k8s clusters run with `ResourceQuota`s, but the deployments created by for the target allocator don't have any set, so they're blocked from creation (at the ReplicaSet level which isn't reported back upwards)

### PR DESCRIPTION
**Pull request by yuriolisa**
Resolve #1103

### CODE REVIEW

**Comment by jaronoff97** [comment]
@yuriolisa seems like the e2e tests need to be updated too here

**Comment by jaronoff97** [comment]
@yuriolisa seems like we're still failing the install:
```
 logger.go:42: 16:42:04 | targetallocator-features/0-install | test step failed 0-install
    logger.go:42: 16:42:04 | targetallocator-features/0-install | collecting log output for [type==pod,label: app.kubernetes.io/component=opentelemetry-targetallocator]
    logger.go:42: 16:42:04 | targetallocator-features/0-install | running command: [kubectl logs --prefix -l app.kubernetes.io/component=opentelemetry-targetallocator -n kuttl-test-prepared-bobcat --all-containers --tail=10]
    logger.go:42: 16:42:04 | targetallocator-features/0-install | No resources found in kuttl-test-prepared-bobcat namespace.
    case.go:361: failed in step 0-install
    case.go:363: statefulsets.apps "stateful-collector" not found
    case.go:363: deployments.apps "stateful-targetallocator" not found
    logger.go:42: 16:42:04 | targetallocator-features | targetallocator-features events from ns kuttl-test-prepared-bobcat:
    logger.go:42: 16:42:04 | targetallocator-features | Deleting namespace: kuttl-test-prepared-bobcat
```

**Comment by yuriolisa** [comment]
> @yuriolisa seems like we're still failing the install:
> 
> ```
>  logger.go:42: 16:42:04 | targetallocator-features/0-install | test step failed 0-install
>     logger.go:42: 16:42:04 | targetallocator-features/0-install | collecting log output for [type==pod,label: app.kubernetes.io/component=opentelemetry-targetallocator]
>     logger.go:42: 16:42:04 | targetallocator-features/0-install | running command: [kubectl logs --prefix -l app.kubernetes.io/component=opentelemetry-targetallocator -n kuttl-test-prepared-bobcat --all-containers --tail=10]
>     logger.go:42: 16:42:04 | targetallocator-features/0-install | No resources found in kuttl-test-prepared-bobcat namespace.
>     case.go:361: failed in step 0-install
>     case.go:363: statefulsets.apps "stateful-collector" not found
>     case.go:363: deployments.apps "stateful-targetallocator" not found
>     logger.go:42: 16:42:04 | targetallocator-features | targetallocator-features events from ns kuttl-test-prepared-bobcat:
>     logger.go:42: 16:42:04 | targetallocator-features | Deleting namespace: kuttl-test-prepared-bobcat
> ```

Right. I thought it was an indentation issue, however, it seems to be on the logic. I'll debug further and submit a fix.

**Review by TylerHelmuth — COMMENTED** [review]
Do we want to set default values [here](https://github.com/open-telemetry/opentelemetry-operator/blob/70e760e4db41bc9108981edaa1505a7cbab661a1/apis/v1alpha1/opentelemetrycollector_webhook.go#L43) like we do for the [autoinstrumentation](https://github.com/open-telemetry/opentelemetry-operator/blob/70e760e4db41bc9108981edaa1505a7cbab661a1/apis/v1alpha1/instrumentation_webhook.go#L55)?

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 1]
```diff
@@ -192,6 +192,9 @@ type OpenTelemetryTargetAllocator struct {
 	// that can be run in a high availability mode is consistent-hashing.
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
+	// Resources to set on the OpenTelemetryTargetAllocator pods.
```
Comment (TylerHelmuth) [inline apis/v1alpha1/opentelemetrycollector_types.go:195]:
```suggestion
	// Resources to set on the OpenTelemetryTargetAllocator containers.
```

File: pkg/targetallocator/container_test.go [hunk 2]
```diff
@@ -77,3 +79,33 @@ func TestContainerVolumes(t *testing.T) {
...
+		},
+	}
+
+	cfg := config.New()
+
+	// test
+	c := Container(cfg, logger, otelcol)
+
+	// verify
+	assert.Equal(t, resource.MustParse("100m"), *c.Resources.Limits.Cpu())
```
Comment (pavolloffay) [inline pkg/targetallocator/container_test.go:107]: just a nit: asserting on the whole `corev1.ResourceRequirements` might be better. I would extract to to a variable and compare it here.

**Review by pavolloffay — APPROVED** [review]

**Comment by yuriolisa** [comment]
> Do we want to set default values [here](https://github.com/open-telemetry/opentelemetry-operator/blob/70e760e4db41bc9108981edaa1505a7cbab661a1/apis/v1alpha1/opentelemetrycollector_webhook.go#L43) like we do for the [autoinstrumentation](https://github.com/open-telemetry/opentelemetry-operator/blob/70e760e4db41bc9108981edaa1505a7cbab661a1/apis/v1alpha1/instrumentation_webhook.go#L55)?

@TylerHelmuth, I don't have objections to setting default values in that case, but how much do you believe would be a good fit?

**Review by jaronoff97 — APPROVED** [review]

**Comment by yuriolisa** [comment]
> @TylerHelmuth i think 250mb of memory and 100m cores is a solid start, for _a lot_ of targets we see it use around 500mb of mem and 200m cores.

Okidoki, I'll add that values to the PR.

**Review by TylerHelmuth — APPROVED** [review]

**Review by frzifus — APPROVED** [review]

**Pull request merged**